### PR TITLE
[Layer] Remove Tensor setDataType() usage

### PR DESCRIPTION
**Pull request by djeong20**
In several layers, there are attempts to change the data type of a Tensor object after initializing it.
This is currently possible but can cause issues down the line (e.g., treat FloatTensor object as HalfTensor).
As such, the setDataType() method will be removed and considered not to be used in future updates.
Instead, users will need to provide the desired data type when creating a new tensor. 

**Self-evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test:   [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

**Review by skykongkong8 — APPROVED** [review]
Think this is quite important point.
I will keep this in mind in the future development of myself as well :)

**Comment by myungjoo** [comment]
Question:

In a network, is the Tensor name required to be unique?
```
  Tensor tensor =
    Tensor("some_name", value.getFormat(), value.getDataType());
```

(I hope not..)

File: nntrainer/layers/layer_normalization_layer.cpp [hunk 1]
```diff
@@ -236,8 +235,8 @@ void LayerNormalizationLayer::calcDerivative(RunLayerContext &context) {
   TensorDim::TensorType weight_tensor_type =
     context.getWeight(wt_idx[LNParams::gamma]).getTensorType();
 
-  Tensor empty;
-  empty.setTensorType(weight_tensor_type);
+  Tensor empty =
+    Tensor("empty", weight_tensor_type.format, weight_tensor_type.data_type);
```
Comment (jijoongmoon) [inline nntrainer/layers/layer_normalization_layer.cpp:239]: We can set the format and data_type with weight_tensor_type. No need to set it. Also, we do not need to set the tensor name always.  I think when we create tensor like here, ```Tensor empty```, then it does not allocate the buffer. So.. I think it is ok to set the Tensor type here. The reason we set like this.. we do not want to allocate a buffer at this time.

**Comment by djeong20** [comment]
> Question:
> 
> In a network, is the Tensor name required to be unique?
> 
> ```
>   Tensor tensor =
>     Tensor("some_name", value.getFormat(), value.getDataType());
> ```
> 
> (I hope not..)

I don't think the name has to be unique.

File: nntrainer/layers/layer_normalization_layer.cpp (hunk 1 above)
Comment (djeong20) [reply nntrainer/layers/layer_normalization_layer.cpp:239]: As we discussed offline, `Tensor empty;` would create FloatTensor as a default. 
I will make an additional Tensor constructor that only takes data type (e.g., `Tensor empty = Tensor(weight_tensor_type.data_type);`)

**Comment by myungjoo** [comment]
> > Question:
> > In a network, is the Tensor name required to be unique?
> > ```
> >   Tensor tensor =
> >     Tensor("some_name", value.getFormat(), value.getDataType());
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > (I hope not..)
> 
> I don't think the name has to be unique.

If this is for debugging and developer referencing, it's ok.

**Review by myungjoo — APPROVED** [review]

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**